packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -1695,4 +1698,121 @@ describeSpec('Listens:', [], () => {
+          .userDeletes('collection/a')
+          // Listen to the query again and verify that the empty snapshot is
+          // raised from cache, even though the write is not yet acknowledged.
+          .userListens(query1, { resumeToken: 'resume-token-1000' })
+          .expectEvents(query1, { fromCache: true })
+      );
+    }
+  );
+
+  specTest(
+    'Empty initial snapshot is raised from cache in multiple tabs',
+    ['multi-client'],
+    () => {
+      const query1 = query('collection');
+      return (
+        client(0, /* withGcEnabled= */ false)
+          // Populate the cache with the empty query results.
+          .userListens(query1)
+          .watchAcksFull(query1, 1000)
+          .expectEvents(query1, { fromCache: false })
+          // Query is shared in second client
+          .client(1)
+          .expectListen(query1)
+          .client(0)
+          .userUnlistens(query1)
+          .watchRemoves(query1)
+          .client(1)
+          .expectUnlisten(query1)
+          // Re-listen to the query in second client and verify that the empty
+          // snapshot is raised from cache.
+          .userListens(query1, { resumeToken: 'resume-token-1000' })
+          .expectEvents(query1, { fromCache: true })
+          .client(0)
+          .expectListen(query1, { resumeToken: 'resume-token-1000' })
+          // Verify that another snapshot is raised once the query result comes
+          // back from Watch.
+          .watchAcksFull(query1, 2000)
+          .client(1)
+          .expectEvents(query1, { fromCache: false })
+      );
+    }
+  );
+  specTest(
+    'Empty-due-to-delete initial snapshot is raised from cache in multiple tabs',
+    ['multi-client'],
+    () => {
+      const query1 = query('collection');
+      const doc1 = doc('collection/a', 1000, { v: 1 });
+      const doc1Deleted = deletedDoc('collection/a', 2000);
+
+      return (
+        client(0, /* withGcEnabled= */ false)
+          // Populate the cache with the empty query results.
+          .userListens(query1)
+          .watchAcksFull(query1, 1000, doc1)
+          .expectEvents(query1, { added: [doc1] })
+          .userUnlistens(query1)
+          .watchRemoves(query1)
+          // Delete the only document in the result set locally on the client.
+          .userDeletes('collection/a')
+          // Re-listen to the query in second client and verify that the empty
+          // snapshot is raised from cache with local mutation.
+          .client(1)
+          .userListens(query1)
+          .expectEvents(query1, { fromCache: true })
+          // Should get events once stream is caught up.
+          .client(0)
+          .expectListen(query1, { resumeToken: 'resume-token-1000' })
+          .writeAcks('collection/a', 2000)
+          .watchAcksFull(query1, 2000, doc1Deleted)
+          .client(1)
+          .expectListen(query1, { resumeToken: 'resume-token-2000' })
```

Comment:
this test is always passing regardless of the resume token in this line(1813). why?
ie: replace line 1813 with any of this: 
```
.expectListen(query1)
.expectListen(query1, { resumeToken: 'resume-token-1000' })
.expectListen(query1, { resumeToken: 'resume-token-2000' })
.expectListen(query1, { resumeToken: 'resume-token-3000' })
```

Reply:
We figured out that you can remove the expectListen() altogether (it shouldn't be there). It looks like a bug in the spec test framework since it's effectively ignoring this call. But, oh well.